[Hong Ming] iP

chm780 opened the pull request:
# Duke your best friend
> "When you have eliminated all which is impossible, then whatever remains, however improbable, must be the truth." ([source](https://www.google.com/search?q=detective+conan&rlz=1C5CHFA_enSG1019SG1019&source=lnms&tbm=isch&sa=X&ved=2ahUKEwikiKuIoOf5AhXR43MBHT8FBCgQ_AUoAXoECAIQAw&biw=1440&bih=821&dpr=2))

Duke
I'm simply too great. Here are reasons why:
- I'm smart
- I'm insane
- I'm kidding

All you need to do is 

1. Chillax [chillllllll](https://www.google.com/search?q=chill&rlz=1C5CHFA_enSG1019SG1019&source=lnms&tbm=isch&sa=X&ved=2ahUKEwiY2MqNoef5AhVmSGwGHSR4BIwQ_AUoAXoECAIQAw&biw=1440&bih=789&dpr=2) :point_left: click it :palm_tree:
2. click that :astonished:
3. learn it :whale: :dragon:
4. haha  :100:

I will just charge you **10 cents** per usage!

Features:
- [x] Become a Math expert
- [ ] Become Jay chou
- [ ] Become a legend

If you are pro, then _try_ it out :smile: !! 

---
```java
public static void main(String[] args) {
    System.out.println("Hi come challenge me!");
}
```

Commit:
Add Gradle support

chm780 committed:
Greet, echo, exit

chm780 committed:
Add, List

chm780 committed:
added Task class

chm780 committed:
added methods to Task

chm780 committed:
Mark as Done features added

chm780 committed:
more elegant prefix check

chm780 committed:
marking unmarking statement within Task class

chm780 committed:
Added subclasses of Tasks

chm780 committed:
Subclasses of Tasks

chm780 committed:
updated subclasses

chm780 committed:
ToDos, Events, Deadlines

chm780 committed:
change name to Dude

chm780 committed:
some edits

chm780 committed:
some edits

chm780 committed:
added DukeException

chm780 committed:
Handling errors

chm780 committed:
added delete

chm780 committed:
added exceptions

chm780 committed:
delete exceptions added

chm780 committed:
minor edit

chm780 committed:
added test cases

chm780 committed:
minor edits

chm780 committed:
remove redundants

chm780 committed:
include save

chm780 committed:
Dates and Times

chm780 committed:
Add data

chm780 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Dates and Times

# Conflicts:
#	src/main/java/Duke.java

chm780 committed:
Use more OOP

chm780 committed:
Organise into Packages

chm780 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

chm780 committed:
Build A-Gradle

chm780 committed:
Add JUnit Tests

chm780 committed:
Fix minor parts

chm780 committed:
Add JavaDoc comments

chm780 committed:
Add a find function

chm780 committed:
Follow the Coding Standard

chm780 committed:
Merge branch 'branch-A-CodingStandard'

chm780 committed:
Merge branch 'branch-Level-9'

chm780 committed:
Fix some bugs

bigcrushes commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,31 @@
+        this.date = date;
+    }
+
+    /**
+     * Add new tasks.
+     *
+     * @param taskList the list of tasks
+     */
```
I noticed this javadoc repeated throughout the AddCommand child classes. Putting it in the parent class and using @inheritDoc is a good alternative!

bigcrushes commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+package duke.command;
+
+import duke.task.Task;
```
Imports should be in alphabetical order

bigcrushes commented on src/main/java/duke/main/FileManipulation.java:
```diff
@@ -0,0 +1,84 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class FileManipulation {
```
I noticed all the public classes do not really have javadocs, it might be useful for some of the classes you've introduced yourself

bigcrushes commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,96 @@
+     */
+    public static Command parse(String input) throws DukeException{
+        String[] userInput = input.trim().split(" ", 2);
+        switch (userInput[0]) {
```
The case blocks should be at the same indentation as the switch block.

bigcrushes reviewed (commented):
Overall the code is clean, easy to read and largely followed the java coding standards

chm780 committed:
Add checkstyle

chm780 committed:
Add GUI to Duke

chm780 committed:
Add GUI

chm780 committed:
Edit GUI

TJun-Jie commented on addressbook-level3-master/src/main/java/seedu/address/MainApp.java:
```diff
@@ -0,0 +1,183 @@
+    protected Model model;
+    protected Config config;
+
+    @Override
```
Can write Java Docs here

TJun-Jie replied:
Can Write javadocs for this function

TJun-Jie commented on addressbook-level3-master/src/main/java/seedu/address/commons/core/Config.java:
```diff
@@ -0,0 +1,63 @@
+    private Level logLevel = Level.INFO;
+    private Path userPrefsFilePath = Paths.get("preferences.json");
+
+    public Level getLogLevel() {
```
Can write java docs for this method

TJun-Jie reviewed (commented):
Great job!

chm780 committed:
Add javadocs

Most of the public methods and classes have not been given javadocs.

Javadocs allows others to understand my methods and classes better.

I add javadocs to public methods and classes.

To comply with the coding standards.

Let's,
* add javadocs to public methods and classes

chm780 committed:
Add assertions

No use of assertions.

Possible errors in program may cause the character to not be 'T', 'D' or
'E'.

Program will assert error should the character is not 'T', 'D' or 'E'.

Only 'T', 'D' and 'E' are recognised as tasks.

Let's,
* Add assertions to handle any errors caused by the programmers

chm780 committed:
Merge pull request #1 from chm252/branch-A-Assertions

Branch a assertions

chm780 committed:
Improve code quality

chm780 committed:
Merge pull request #2 from chm252/branch-A-CodeQuality

Improve code quality

chm780 committed:
Add undo

chm780 committed:
Make GUI better and tidy javadocs

chm780 committed:
Add a User Guide

chm780 committed:
Add screenshot

chm780 committed:
Fix checkstyle error

chm780 committed:
Fix bugs

chm780 committed:
Remove redundant file

chm780 committed:
Fix bugs
